### ISSUE TITLE
Layout/MultilineMethodCallIndentation reports false positives if using Ruby 2.7 numbered parameters

### ISSUE THREAD

**Issue by DMA57361** [rubocop-hq/rubocop#7779]
## Expected behavior

Code that correctly confirms to the Layout/MultilineMethodCallIndentation cop's default "aligned" enforced style while making use of blocks with Ruby 2.7's new numbered parameters feature should not raise an offense.

## Actual behavior

Lines that follow the numbered parameter block are reported as being indented incorrectly.

## Steps to reproduce the problem

Given the following `example.rb` file:
``` ruby
File.read('data.yml')
    .then { |file| YAML.safe_load file } # normal block param
    .transform_values(&:downcase)
    .freeze

File.read('data.yml')
    .then { YAML.safe_load _1 } # numbered param
    .transform_values(&:downcase)
    .freeze
```

Running rubocop cop gives the following results:
```
$ rubocop example.rb --only Layout/MultilineMethodCallIndentation
Inspecting 1 file
C

Offenses:

example.rb:8:5: C: Layout/MultilineMethodCallIndentation: Use 2 (not 4) spaces for indenting an expression spanning multiple lines.
    .transform_values(&:downcase)
    ^^^^^^^^^^^^^^^^^
example.rb:9:5: C: Layout/MultilineMethodCallIndentation: Use 2 (not 4) spaces for indenting an expression spanning multiple lines.
    .freeze
    ^^^^^^^

1 file inspected, 2 offenses detected
```

Note how the first version of the code in the file (using normal block params) is considered correct while the second version (using Ruby 2.7 numbered params) is considered incorrect.

## RuboCop version

```
$ bundle exec rubocop -V
0.80.1 (using Parser 2.7.0.4, running on ruby 2.7.0 x86_64-linux)
```
